pkg/actors/actor_error.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package actors
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+
+	invokev1 "github.com/dapr/dapr/pkg/messaging/v1"
+)
+
+type ActorError struct {
+	body        []byte
+	headers     invokev1.DaprInternalMetadata
+	contentType string
+	statusCode  int
+	message     string
+}
+
+func NewActorError(invokeResponse *invokev1.InvokeMethodResponse) *ActorError {
```

Comment:
I think this should return `error` so you can return other errors in case they are not ActorError. See my other comments

Reply:
I like the idea. Done.